### PR TITLE
[DA-215] Use the new unified configuration file

### PR DESCRIPTION
**Pull request by thescouser89**

### CODE REVIEW

**Comment by thescouser89** [comment]
Please don't merge right now. The unified config file is still in proposal and hasn't been merged yet in the main pnc configuration. The testsuite will fail, and I will adjust it once the unified config file is accepted.

We'll assume that the unified config file has the name 'main-config.json'

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

File: common/src/main/java/org/jboss/da/common/util/Configuration.java [hunk 1]
```diff
@@ -18,7 +24,25 @@ public DAConfig getConfig() throws ConfigurationParseException {
 
         try (InputStream configStream = getConfigStream()) {
             ObjectMapper mapper = new ObjectMapper();
-            return mapper.readValue(configStream, DAConfig.class);
+
+            // Note: I need to add the dummy classes GlobalGroupWrapper and
+            // PNCGroupWrapper since the version of Jackson we are using
+            // doesn't support FAIL_ON_INVALID_SUBTYPE (it's in v2.2)
+            mapper.registerSubtypes(new NamedType(GlobalGroupWrapper.class, "global"));
+            mapper.registerSubtypes(new NamedType(PNCGroupWrapper.class, "pnc"));
```
Comment (janinko) [inline common/src/main/java/org/jboss/da/common/util/Configuration.java]: Did you tried to use Default implementation? Or codehause jackon doesn't support it?

Comment (thescouser89) [inline common/src/main/java/org/jboss/da/common/util/Configuration.java]: @janinko what would be the default implementation?

**Comment by jbartece** [comment]
:+1:

**Pull request merged**